cssselect2/parser.py

```diff
@@ -266,14 +267,18 @@ def skip_whitespace(self):
             if peek is None or peek.type != 'whitespace':
                 break
             self.next()
+            if u'\n' in peek.value:
+                self.lines += 1
+
             has_whitespace = True
-        return has_whitespace
+        return has_whitespace, self.lines
 
 
 class Selector(object):
-    def __init__(self, tree, pseudo_element=None, extensions=None):
+    def __init__(self, tree, pseudo_element=None, extensions=None, src_line=0):
         self.parsed_tree = tree
         self.extensions = extensions
+        self.source_line = src_line
```

Comment:
👍  for `source_line_offset` to clarify that it is an offset from the source line (or I guess the other one works too). 

💭  What if you did the offset arithmetic in the parser and just made the field be named `source_line`? Then, `source_line` always means the line in the source code and if a user wanted to calculate the offset they could do the arithmetic themselves, no?